lib/ansible/modules/cloud/ovirt/ovirt_vnic_profile.py

```diff
@@ -171,26 +171,26 @@ def _get_network_id(self):
 
     def _get_qos_id(self):
         qoss_service = self._get_dcs_service().service(self._get_dcs_id()).qoss_service()
-        return get_id_by_name(qoss_service, self.param('qos'))
+        return get_id_by_name(qoss_service, self.param('qos')) if self.param('qos') else None
 
     def _get_network_filter_id(self):
         nf_service = self._connection.system_service().network_filters_service()
         return get_id_by_name(nf_service, self.param('network_filter')) if self.param('network_filter') else None
 
     def _get_network_filter(self):
         network_filter = None
-        if self.param('network_filter'):
-            network_filter = otypes.NetworkFilter(id=self._get_network_filter_id())
-        elif self.param('network_filter') == '' or self.param('pass_through') == 'enabled':
+        if self.param('network_filter') == '' or self.param('pass_through') == 'enabled':
             network_filter = otypes.NetworkFilter()
+        elif self.param('network_filter'):
+            network_filter = otypes.NetworkFilter(id=self._get_network_filter_id())
```

Comment:
Please add a comment to explain why the order is important.

Reply:
The issue was when specified network_filter and pass_through enabled  it would go through first if to create network filter. But now when there is check which will automatically raise exeption. Should i put it back as it was?

Reply:
You can leave it as is, but add a comment which explains that the order is important